examples/preprocessing/plot_smoothing_noise_quantile_transform.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+========================================================
+Effect of smoothing noise when using QuantileTransformer
+========================================================
+
+This example shows the effect of applying a small noise before
+computing the quantiles in the QuantileTransformer. The parameter
+``smoothing_noise`` can be used if a constant feature value is
+predominant in the dataset. It will lead to a difficult interpretation
+when the quantiles computed are manually checked.
+
+"""
+
+# Author:  Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.preprocessing import QuantileTransformer
+
+print(__doc__)
+
+N_QUANTILES = 1000
+FEAT_VAL = 3.0
+
+
+def plot_transform_feat_val(ax, transformer, title):
+    """Plot the full transformation mapping the feature values as well as
+    a single feature."""
+    ref = np.linspace(0, 1, num=N_QUANTILES)
+
+    ax.plot(transformer.quantiles_, ref)
+    ax.scatter(FEAT_VAL, transformer.transform(FEAT_VAL), c='r',
+               label=r'$f({0}) = {1:.2f}$'.format(
+                   FEAT_VAL,
+                   np.ravel(transformer.transform(FEAT_VAL))[0]))
+    ax.set_xlabel('Feature values')
+    ax.set_ylabel('Quantiles in %')
+    ax.set_title(title)
+    ax.legend(loc=4)
+    # make nice axis layout
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.set_xlim([1, 5.1])
+    ax.set_ylim([0, 1])
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+
+
+###############################################################################
+# We can create a synthetic dataset representing the customers'
+# ratings for a restaurant. The scale used is ranging from 1 to 5 and
+# a large number of customers attributed a grade of 3 to the current
+# restaurant.
+
+X = np.array([1] * 1000 +
+             [2] * 2000 +
```

Comment:
Maybe you could switcht the counts for 1 star and 2 stars: 2000 x 1 star and 1 x 2 stars so as to break the vertical symmetry of the plot.